src/zenml/utils/pipeline_docker_image_builder.py

```diff
@@ -437,8 +443,32 @@ def gather_requirements_files(
             - Packages installed in the local Python environment
             - Requirements defined by stack integrations
             - Requirements defined by user integrations
+            - Requirements exported from a pyproject.toml
             - User-defined requirements
         """
+        implicit_requirements = False
+        pyproject_path = docker_settings.pyproject_path
+        requirements = docker_settings.requirements
+
+        if not any(
+            [
+                docker_settings.replicate_local_python_environment,
+                docker_settings.required_integrations,
```

Comment:
Does this mean that adding an extra integration (or a package) to be installed disables the discovery of the implicit requirements? So, if I was not using any of these settings before and now want to add `sklearn` to my docker settings, suddenly the `requirements.txt` will not be detected. I am not sure if this is the optimal behavior here.

Reply:
Correct. I'm not super convinced by it myself to be honest, but what would be a better alternative in your opinion?

Reply:
Yeah, I guess I do not have a super clear alternative either. 

The reason why I was a bit hesitant here was this. We have a default behavior that decides whether we look for these files or not. By adding an additional integration, (that I intuitively thought would not change this behavior), I change the way that it looks for these files. I guess the `replicate_local_python_environment` flag at least allows people to control the behavior. Perhaps we can write a sentence about this in the docs?